### PR TITLE
Dev swig

### PR DESCRIPTION
**Pull request by michaeltryby**
Part of work needed to address Issue #110. These are the changes I'm proposing to facilitate SWIG wrapping EPANET. They are not ready to merge yet, but I would appreciate your review. Thanks.

### CODE REVIEW

File: src/rules.c [hunk 1]
```diff
@@ -114,7 +114,7 @@ int takeactions(EN_Project *pr);
 void clearactlist(rules_t *rules);
 void clearrules(EN_Project *pr);
 void ruleerrmsg(EN_Project *pr, int);
-int writeRuleinInp(EN_Project *pr, FILE *f, int RuleIdx);
+//int writeRuleinInp(EN_Project *pr, FILE *f, int RuleIdx);
```
Comment (samhatchett) [inline src/rules.c]: was this not needed?

**Comment by michaeltryby** [comment]
Yeah it was necessary. I moved the declaration to funcs.h to remove a compiler warning on msvc.

**Comment by michaeltryby** [comment]
void *EN_ProjectHandle was created and substituted for EN_Project. Code was added to each API function to cast the handle to a project so it can be used within the function. The changes were compiled using MSVC and MinGW. Minor changes were made to eliminate compiler warnings.

**Comment by michaeltryby** [comment]
@samhatchett Next I was going to add a simple error manager to facilitate automatically throwing an exception in the wrapper code . Any thoughts/concerns about that?

**Comment by michaeltryby** [comment]
@samhatchett this will involve creating a variable to hold the error state and providing an API to clear it and to check it. This makes it easy to have SWIG insert custom exception handling into the wrapper. 

Adding this API will make `EN_getError()` redundant, but I suppose we should leave it for backwards compatibility. Thoughts?

**Pull request closed without merging**